### PR TITLE
Feature storages

### PR DESCRIPTION
**Pull request by EanCuznaivy**

### CODE REVIEW

File: AElf.Kernel/Storages/ChangesStore.cs [hunk 1]
```diff
@@ -1,24 +1,64 @@
...
+        private readonly KeyValueDatabase _keyValueDatabaseForPaths;
 
-        public ChangesStore(IKeyValueDatabase keyValueDatabase)
+        private static readonly Hash HashToGetPaths = new Hash("paths".CalculateHash());
+
+        public ChangesStore(KeyValueDatabase keyValueDatabase, KeyValueDatabase keyValueDatabaseForPaths)
         {
             _keyValueDatabase = keyValueDatabase;
+            _keyValueDatabaseForPaths = keyValueDatabaseForPaths;
+            _keyValueDatabaseForPaths.SetAsync(HashToGetPaths, new List<Hash>());
```
Comment (EanCuznaivy) [inline AElf.Kernel/Storages/ChangesStore.cs]: @loning I'm not sure whether List<Hash> can be serialized by protobuf, but for now the implementation need to get all the changes to generate a WorldState instance, thus I use another KeyValueDatabase to store the hash list. So can you help to review this implementation?

Comment (loning) [reply AElf.Kernel/Storages/ChangesStore.cs]: you can define a protobuf message with repeat filed

Comment (loning) [reply AElf.Kernel/Storages/ChangesStore.cs]: see BlcokBody -> transactions

File: AElf.Kernel.Tests/DataProviderTest.cs [hunk 2]
```diff
@@ -1,10 +1,77 @@
...
+            var keys = GenerateKeys(setList).ToList();
+            
+            var chain = new Chain(Hash.Generate());
+            var chainManager = new ChainManager(_chainStore);
+            await chainManager.AddChainAsync(chain.Id);
+            var hash = Hash.Generate();
+
+            var address = Hash.Generate();
+            var accountContextService = new AccountContextService();
+            var worldStateManager = new WorldStateManager(_worldStateStore, hash, accountContextService,
```
[the remark this answers is not in the document]
Comment (EanCuznaivy) [reply AElf.Kernel.Tests/DataProviderTest.cs]: @loning Then I have to modify the params of IWorldStateManager's method GetAccountDataProvider and SetWorldStateToCurrentState defined before, adding a parameter "Hash preBlockHash", is this okay?

**Pull request merged**